builtin/credential/cert/path_certs.go

```diff
@@ -147,6 +169,47 @@ func (b *backend) pathCertWrite(
 	policies := policyutil.ParsePolicies(d.Get("policies"))
 	allowedNames := d.Get("allowed_names").([]string)
 
+	// Parse the ttl (or lease duration)
+	systemDefaultTTL := b.System().DefaultLeaseTTL()
+	ttl := time.Duration(d.Get("ttl").(int)) * time.Second
+	if ttl == 0 {
+		ttl = time.Second * time.Duration(d.Get("lease").(int))
+	}
+	if ttl > systemDefaultTTL {
+		return logical.ErrorResponse(fmt.Sprintf("Given ttl of %d seconds is greater than current mount/system default of %d seconds", ttl/time.Second, systemDefaultTTL/time.Second)), nil
+	}
+
+	// Parse max_ttl
+	systemMaxTTL := b.System().MaxLeaseTTL()
+	maxTTL := time.Duration(d.Get("max_ttl").(int)) * time.Second
+	if maxTTL > systemMaxTTL {
+		return logical.ErrorResponse(fmt.Sprintf("Given max_ttl of %d seconds is greater than current mount/system default of %d seconds", maxTTL/time.Second, systemMaxTTL/time.Second)), nil
+	}
+
+	// Logic pertaining to interaction between ttl and max_ttl
+	if maxTTL != 0 {
+		// If ttl is 0, set the ttl as the max ttl
+		if ttl == 0 {
+			ttl = maxTTL
+		}
+
+		// Error if ttl is larger than maxTTL
+		if ttl > maxTTL {
+			return logical.ErrorResponse("ttl should be shorter than max_ttl"), nil
+		}
+	}
+
+	// Parse period
+	period := time.Duration(d.Get("period").(int)) * time.Second
+	if period > systemMaxTTL {
+		return logical.ErrorResponse(fmt.Sprintf("Given period of %d seconds is greater than the backend's maximum TTL of %d seconds", period/time.Second, systemMaxTTL/time.Second)), nil
+	}
+
+	// If a period is provided, use that as the ttl instead, ignoring previously set ttl
+	if period > 0 {
+		ttl = period
```

Comment:
Also don't do this. Set according to period at token issue or renew time. Don't modify the input parameters.